src/Interpreters/TreeRewriter.cpp

```diff
@@ -976,7 +976,8 @@ TreeRewriterResultPtr TreeRewriter::analyzeSelect(
     result.required_source_columns_before_expanding_alias_columns = result.required_source_columns.getNames();
 
     /// rewrite filters for select query, must go after getArrayJoinedColumns
-    if (settings.optimize_respect_aliases && result.metadata_snapshot)
+    bool is_initiator = getContext()->getClientInfo().distributed_depth == 0 || settings.distributed_group_by_no_merge > 0;
```

Comment:
I'm sorry that I didn't make it clear. `getContext()->getClientInfo().distributed_depth == 0 ` is enough. `settings.distributed_group_by_no_merge > 0` is used for other purposes and should not be applied here (it's wrong)